core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java

```diff
@@ -289,6 +295,110 @@ public AbstractBindingBuilder produces(Iterable<MediaType> produceTypes) {
+     * @see ConditionalParam
+     */
+    public AbstractBindingBuilder matchesParams(Iterable<String> paramPredicates) {
+        this.paramPredicates = RoutingPredicate.copyOfParamPredicates(
+                requireNonNull(paramPredicates, "paramPredicates"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link Route} to accept a request if it the specified {@code predicate} returns {@code true}.
+     */
+    public AbstractBindingBuilder matchesParams(RoutingPredicate<HttpParameters> predicate) {
+        paramPredicates = ImmutableList.of(requireNonNull(predicate, "predicate"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link Route} to accept a request if it matches all the specified predicates for
+     * {@link HttpHeaders}. The predicate can be one of the following forms:
+     * <ul>
+     *     <li>{@code some-header=some-value} which means that the request must have a
+     *     {@code some-header: some-value} header</li>
+     *     <li>{@code some-header!=some-value} which means that the request must not have a
+     *     {@code some-header: some-value} header</li>
+     *     <li>{@code some-header} which means that the request must contain a {@code some-header} header</li>
+     *     <li>{@code !some-header} which means that the request must not contain a {@code some-header}
+     *     header</li>
+     * </ul>
+     *
+     * @see ConditionalHeader
+     */
+    public AbstractBindingBuilder matchesHeaders(String... headerPredicates) {
+        return matchesHeaders(ImmutableList.copyOf(requireNonNull(headerPredicates, "headerPredicates")));
+    }
+
+    /**
+     * Sets the {@link Route} to accept a request if it matches all the specified predicates for
+     * {@link HttpHeaders}. The predicate can be one of the following forms:
+     * <ul>
+     *     <li>{@code some-header=some-value} which means that the request must have a
+     *     {@code some-header: some-value} header</li>
+     *     <li>{@code some-header!=some-value} which means that the request must not have a
+     *     {@code some-header: some-value} an header</li>
+     *     <li>{@code some-header} which means that the request must contain a {@code some-header} header</li>
+     *     <li>{@code !some-header} which means that the request must not contain a {@code some-header}
+     *     header</li>
+     * </ul>
+     *
+     * @see ConditionalHeader
+     */
+    public AbstractBindingBuilder matchesHeaders(Iterable<String> headerPredicates) {
+        this.headerPredicates = RoutingPredicate.copyOfHeaderPredicates(
+                requireNonNull(headerPredicates, "headerPredicates"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link Route} to accept a request if it the specified {@code predicate} returns {@code true}.
+     */
+    public AbstractBindingBuilder matchesHeaders(RoutingPredicate<HttpHeaders> predicate) {
+        headerPredicates = ImmutableList.of(requireNonNull(predicate, "predicate"));
```

Comment:
Perhaps we should add to the existing list, so that a user can specify multiple `RoutingPredicate`s fluently? e.g. `builder.matchesHeaders(pred1).matchesHeaders(pred2)`

Reply:
That's what I wanted at first. But the other API doesn't do that now. So I made it like this. Is it okay to make this method as you suggested? :-)